### DISCUSSION TITLE
Deploy new resource based on Docker image

### DISCUSSION THREAD

**Discussion by watzon** [coollabsio/coolify#1293] [Feature Requests]
Currently we have the ability to deploy resources base on a git repo, Dockerfile, or docker-compose file. The one thing lacking seems to be the ability to deploy a resource simply using a hosted image (via docker hub, ghcr, etc). It would be really nice, since doing it via Dockerfile can be kind of annoying and not as intuitive.

**Comment by endquote** [comment]
I've wondered this too, a dockerhub path and a bunch of environment variables would often do the trick. I think the intent is that you can do the same thing with a simple compose file though.

**Reply by watzon** [reply]
Exactly my thoughts. Some services like mongo, or any containers you build and publish yourself are likely to be pretty simple and could be set up with just the image tag and some variables. I switched to Coolify from CapRover and it has a really simple way to do this.

**Comment by andrasbacsai** [comment]
I can make it available, but it won't be beginner-friendly, as you need to know which envs you need to set.

**Reply by watzon** [reply]
Tbh I feel like it's still more beginner friendly than the current solution. It might be cool to have some other options for that kind of resource too. Maybe taking some inspiration from Portainer?
![image](https://github.com/coollabsio/coolify/assets/4535422/3898f1c4-9ecd-4ab6-a0cb-a0648a0d7d9c)

It would be cool to have such fine grained control without having to result to a docker-compose file. I'm a sucker for nice UIs.

**Reply by andrasbacsai** [reply]
Yes, that is what I would like to achieve for a dockerfile or docker image-based deployment. 🙂

For Nixpack-based, the only thing I would see that makes sense to add is custom labeling (for Traefik).

**Reply by andrasbacsai** [reply]
I will add a simple way to start any public image from any Docker repositories in the next version and extend its features later on.

**Reply by andrasbacsai** [reply]
It is available in beta.72+
